jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -973,8 +973,12 @@ def _update_stretch_histogram(self, msg={}):
                 x_max = x_limits.max()
                 y_min = max(y_limits.min(), 0)
                 y_max = y_limits.max()
-
                 arr = comp.data[y_min:y_max, x_min:x_max]
+                if self.config == "imviz":

```

Comment:
It assumes 2D, so yes?

Reply:
does it have to assume 2D (can we do this for large cubes as well)?  And if so, maybe we can just check the data shape so that this will be available to image viewers in non-imviz configurations?

Reply:
Is large cube a problem? The histogram only compute for the current slice, right? It is very seldom to have very large slice dimension, at least for the use cases we have so far?

Reply:
The histogram should show whatever is being used to compute stretch percentiles, in my opinion. We're soon allowing toggling between current slice and full cube, so I'd expect the histogram to follow that same choice in the future.

Reply:
I am torn on this because it sounds like the ultimate goal is you want true random sampling from glue-core (see https://github.com/spacetelescope/jdaviz/pull/2735#discussion_r1508260792). This means my fix here would be temporary and the immediate problem I am fixing is large image in Imviz, so the more I localize this now, the easier it is to revert in the future when the actual fix you want is implemented.

Reply:
ok, fair, but I meant it should at least represent the same underlying data (even if we don't do exactly the same right now).  I think this will eventually need to be generalized to the cube case, so if that's easy to do now, great, otherwise I guess we can do it later.

Reply:
We discussed offline and agreed to confine this to Imviz for now since there isn't any reported problem for Cubeviz yet.